src/twisted/web/_element.py

```diff
@@ -58,13 +55,24 @@ def frob(self, a, b):
         if not funcObjs:
             raise TypeError("expose() takes at least 1 argument (0 given)")
         for fObj in funcObjs:
-            fObj.exposedThrough = getattr(fObj, "exposedThrough", [])
-            fObj.exposedThrough.append(self)
+            exposedThrough = getattr(fObj, "exposedThrough", [])
+            exposedThrough.append(self)
+            setattr(fObj, "exposedThrough", exposedThrough)
```

Comment:
If this is to avoid a "no such attribute error" from Mypy, I wonder whether we prefer a `type: ignore` (which notes we are intentionally overriding the type system) is better. @glyph ^^

My thinking is that `type: ignore` invites occasional scrutiny in the future, and maybe that's worthwhile?

Reply:
I was going to argue that `getattr()` allows type checking of the attribute, but I see that I didn't actually annotate `exposedThrough`. I think that's the cleaner solution though: `type: ignore` will make the warning go away but will also prevent mypy from checking the code.

I'll add an annotation of `exposedThrough`. If it is later decided that `type: ignore` is preferred, that's fine too.